packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -352,3 +323,52 @@ class NetworkController extends DisposableController
     }
   }
 }
+
+class CurrentNetworkRequests {
+  CurrentNetworkRequests({required this.onDataUpdate});
+
+  Map<String, NetworkRequest> requestsMapping = <String, NetworkRequest>{};
+  VoidCallback onDataUpdate; // TODO: Callback naming rules
+
+  void updateOrAdd(
+    HttpProfileRequest request,
+    int timelineMicrosOffset,
+  ) {
+    final wrapped = DartIOHttpRequestData(
+      timelineMicrosOffset,
+      request,
+      requestFullDataFromVmService: false,
+    );
+    if (!requestsMapping.containsKey(request.id)) {
+      requestsMapping[wrapped.id] = wrapped;
+    } else {
+      // If we override an entry that is not a DartIOHttpRequestData then that means
+      // the ids of the requestMapping entries may collide with other types
+      // of requests.
+      assert(requestsMapping[request.id] is DartIOHttpRequestData);
+      (requestsMapping[request.id] as DartIOHttpRequestData).merge(wrapped);
+    }
+  }
+
+  void updateWebSocketRequests(
+    List<SocketStatistic> sockets,
+    int timelineMicrosOffset,
+  ) {
+    for (final socket in sockets) {
+      final webSocket = WebSocket(socket, timelineMicrosOffset);
+
+      if (requestsMapping.containsKey(webSocket.id)) {
+        // If we override an entry that is not a Websocket then that means
+        // the ids of the requestMapping entries may collide with other types
+        // of requests.
+        assert(requestsMapping[webSocket.id] is WebSocket);
+      }
+
+      requestsMapping[webSocket.id] = webSocket;
```

Comment:
can you comment to explain why there is no concept of "merge" for web socket requests like we have for http requests